Fails for me on macos 10.14

```
======================================================================
FAIL: test_filter_callback (test_functionality.BasicUsage)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_functionality.py", line 198, in test_filter_callback
    self.assert_traces_almost_equal(r1, stats)
  File "tests/utils.py", line 56, in assert_traces_almost_equal
    self.assert_almost_equal(ttot_orig, t.ttot, err_msg=tline)
  File "tests/utils.py", line 87, in assert_almost_equal
    neg_threshold, y, pos_threshold, err_msg
AssertionError: 0.16028 <= 0.343271 <= 0.28049 is not True. [tests/test_functionality.py:98 a      2      0.000000  0.200350  0.100175]

======================================================================
FAIL: test_tagging_walltime (test_tags.MultiThreadTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_tags.py", line 50, in test_tagging_walltime
    self.assert_traces_almost_equal(t1, traces)
  File "tests/utils.py", line 56, in assert_traces_almost_equal
    self.assert_almost_equal(ttot_orig, t.ttot, err_msg=tline)
  File "tests/utils.py", line 87, in assert_almost_equal
    neg_threshold, y, pos_threshold, err_msg
AssertionError: 1.6032471999999998 <= 3.277530999999999 <= 2.8056826 is not True. [..p/yappi/tests/utils.py:134 burn_io  20     0.000638  2.004059  0.100203]

======================================================================
FAIL: test_simple_tagging (test_tags.SingleThreadTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_tags.py", line 169, in test_simple_tagging
    self.assert_traces_almost_equal(t1, traces)
  File "tests/utils.py", line 56, in assert_traces_almost_equal
    self.assert_almost_equal(ttot_orig, t.ttot, err_msg=tline)
  File "tests/utils.py", line 87, in assert_almost_equal
    neg_threshold, y, pos_threshold, err_msg
AssertionError: 0.1665168 <= 0.33278699999999994 <= 0.2914044 is not True. [..p/yappi/tests/utils.py:134 burn_io  2      0.000000  0.208146  0.104073]

======================================================================
FAIL: test_basic_old_style (test_gevent.SingleThreadTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_gevent.py", line 208, in test_basic_old_style
    self.assert_traces_almost_equal(r1, stats)
  File "tests/utils.py", line 56, in assert_traces_almost_equal
    self.assert_almost_equal(ttot_orig, t.ttot, err_msg=tline)
  File "tests/utils.py", line 87, in assert_almost_equal
    neg_threshold, y, pos_threshold, err_msg
AssertionError: 1.2832392000000001 <= 2.327836 <= 2.2456686 is not True. [..p/yappi/tests/test_asyncio.py:43 a  2      0.000118  1.604049  0.802024]

======================================================================
FAIL: test_basic_old_style (test_asyncio.SingleThreadTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_asyncio.py", line 65, in test_basic_old_style
    self.assert_traces_almost_equal(r1, stats)
  File "tests/utils.py", line 56, in assert_traces_almost_equal
    self.assert_almost_equal(ttot_orig, t.ttot, err_msg=tline)
  File "tests/utils.py", line 87, in assert_almost_equal
    neg_threshold, y, pos_threshold, err_msg
AssertionError: 1.2832392000000001 <= 2.295179 <= 2.2456686 is not True. [..p/yappi/tests/test_asyncio.py:43 a  2      0.000118  1.604049  0.802024]

----------------------------------------------------------------------
Ran 97 tests in 41.206s

FAILED (failures=5, skipped=3)
```

This looks like another timing based test, which are notoriously flakey.

Wonder if we could find a sneaky way to disable the timing checks without disabling the surrounding tests.

--- a/pkgs/development/python-modules/yappi/default.nix
+++ b/pkgs/development/python-modules/yappi/default.nix
@@ -1,18 +1,22 @@
-{ lib, buildPythonPackage, fetchPypi, isPy27, nose }:
+{ lib, buildPythonPackage, fetchFromGitHub, gevent, isPy27, python }:
 
 buildPythonPackage rec {
   pname = "yappi";
-  version = "1.2.5";
+  version = "1.3.0";
   disabled = isPy27; # invalid syntax
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "ad5fa4caf2859e480ffc4ec3e85615a6f7dea852c8f035f2db723f824ed4ba11";
+  src = fetchFromGitHub {
+    owner = "sumerc";
+    repo = pname;
+    rev = "30f94024a0e2e4fa21c220de6a0dc97b4cb2c319";
+    sha256 = "1kvwl3y3c2hivf9y2x1q1s8a2y724iwqd1krq6ryvsbg3inyh8qw";
   };
 
   patches = [ ./tests.patch ];
-
-  checkInputs = [ nose ];
+  checkInputs = [ gevent ];
+  checkPhase = ''
+    ${python.interpreter} run_tests.py
+  '';
 
   meta = with lib; {
     homepage = "https://github.com/sumerc/yappi";

--- a/pkgs/development/python-modules/yappi/tests.patch
+++ b/pkgs/development/python-modules/yappi/tests.patch
@@ -1,12 +1,14 @@
 --- a/tests/test_functionality.py
 +++ b/tests/test_functionality.py
-@@ -74,2 +74,3 @@ class BasicUsage(utils.YappiUnitTestCase):
-    
-+    @_unittest.skip('wall-clock-time-sensitive')
+@@ -262,1 +262,2 @@ class BasicUsage(utils.YappiUnitTestCase):
++    @unittest.skip('wall-clock-time-sensitive')
      def test_get_clock(self):
+@@ -363,2 +363,3 @@ class BasicUsage(utils.YappiUnitTestCase):
+     @unittest.skipIf(os.name == "nt", "do not run on Windows")
++    @unittest.skip('broken-path')
+     def test_run_as_script(self):
 --- a/tests/test_hooks.py
 +++ b/tests/test_hooks.py
-@@ -124,2 +124,3 @@ class ContextIdCallbackTest(utils.YappiUnitTestCase):
- 
+@@ -127,1 +127,2 @@ class ContextIdCallbackTest(utils.YappiUnitTestCase):
 +    @unittest.skip('wall-clock-time-sensitive')
      def test_pause_resume(self):

